### ISSUE TITLE
ExpectReply and ExpectReplyToOriginator do not throw exception on ICommand

### ISSUE THREAD

**Issue by gamblen** [Particular/NServiceBus.Testing#97]
If a message is an ICommand rather than IMessage and is used in Reply or ReplyToOriginator then you get the expected exception:

*System.Exception: Reply is neither supported for Commands nor Events. Commands should be sent to their logical owner. Events should be published.*

This is not raised during testing when using ExpectReplyToOriginator and ExpectReply for a Saga and ExpectReply for Handler.

I have attached a project that fails when you run it, but the unit tests all pass.
[NServiceBusReplyToOriginatorError.zip](https://github.com/Particular/NServiceBus.Testing/files/618870/NServiceBusReplyToOriginatorError.zip)

**Comment by timbussmann** [comment]
Hey @gamblen 

The testing framework does indeed not verify best practices at the moment. This is a special feature which hooks itself into the NServiceBus pipeline to throw in case you're not following the recommended best practices. Since the testing framework doesn't really execute the same pipeline as the production code, best practices aren't verified.

We decided [here](https://github.com/Particular/NServiceBus.Testing/issues/37) that we would not implement this in version 5 but it's still an option as a new feature for version 6. What version of NServiceBus are you currently using?

**Comment by gamblen** [comment]
Hi @timbussmann

I assumed as such, as after I raised the issue I thought I would be a good citizen and see if it was just overlooked in the code, which it isn't as you say.

We are using v6.

**Comment by timbussmann** [comment]
We did not implement that behavior back then because we weren't sure how much value those best practices really provide (as most users probably don't want to specify their conventions in case they are using unobtrusive message conventions). 

I raised a new issue to discuss that feature in general as we not have at least proof of users seeing value in it 👍 If you're interested, we definitely appreciate opinions and thoughts on that topic on the raised issue (#98). @gamblen Are you good with me closing this issue and moving further discussions to the more general "best practice validation feature" issue?
